TutorLizard.BusinessLogic/Services/TutorService.cs

```diff
@@ -131,4 +134,76 @@ public async Task<UnacceptScheduleItemRequestResponse> UnacceptScheduleItemReque
             Success = true
         };
     }
+
+    public async Task<TutorsPendingAdRequestsResponse> ViewAllPendingAdRequests(TutorsPendingAdRequestsRequest request)
+    {
+        List<AdRequestsListDto> adRequests = await _adRequestRepository.GetAll()
+            .Where(adrequest => 
+                adrequest.Ad.TutorId == request.TutorId &&
+                adrequest.ReviewDate == null &&
+                adrequest.IsAccepted == false)
+            .Include(adrequest => adrequest.Ad)
+            .ThenInclude(ad => ad.Category)
+            .Select(adrequest => new AdRequestsListDto(adrequest.Id,
+                                                       adrequest.StudentId,
+                                                       adrequest.AdId,
+                                                       adrequest.IsAccepted,
+                                                       adrequest.Message,
+                                                       adrequest.ReplyMessage,
+                                                       adrequest.IsRemote,
+                                                       adrequest.Ad.Title,
+                                                       adrequest.Ad.Subject,
+                                                       adrequest.Ad.Category.Name))
+            .ToListAsync();
+
+        TutorsPendingAdRequestsResponse response = new()
+        {
+            AdRequests = adRequests
+        };
+
+        return response;
+    }
+
+    public async Task<UpdateTutorsPendingAdRequestResponse> UpdateAdRequest(UpdateTutorsPendingAdRequestRequest request)
+    {
+        var adRequest = await _adRequestRepository
+            .Update(request.AdRequestId, entity =>
+            {
+                 entity.ReviewDate = DateTime.Now;
+                 entity.ReplyMessage = request.ReplyMessage;
+            });
+
+        if (request.Action == UpdateTutorsPendingAdRequestRequest.UpdateAction.Accept)
+        {
+            adRequest = await _adRequestRepository
+                .Update(request.AdRequestId, entity => entity.IsAccepted = true);
+        }
+
+        if (adRequest is null) {
+            UpdateTutorsPendingAdRequestResponse updateFailed = new()
+            {
+                IsSuccessful = false
+            };
+            
+            return updateFailed;
+        }
+
+        UpdateTutorsPendingAdRequestResponse response = new()
+        {
+            UpdatedAdRequestDto =
+            {
```

Comment:
That's not a correct way to do it. You have to give this field a new instance of `AdRequestDto` (`new()` before `{`, but you'd also need to add an empty constructor to the DTO). Right now you try to set values of properties on a null and I get `NullReferenceException` every time.
You could fix it, but again, you are not using this data even once, so maybe just remove this property from response?